airflow/providers/google/cloud/operators/datastore.py

```diff
@@ -191,3 +199,287 @@ def execute(self, context):
             raise AirflowException('Operation failed: result={}'.format(result))
 
         return result
+
+
+class CloudDatastoreAllocateIdsOperator(BaseOperator):
+    """
+    Allocate IDs for incomplete keys. Return list of keys.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:CloudDatastoreAllocateIdsOperator`
+
+    .. seealso::
+        https://cloud.google.com/datastore/docs/reference/rest/v1/projects/allocateIds
+
+    :param partial_keys: a list of partial keys.
+    :type partial_keys: list
+    :param project_id: Google Cloud Platform project ID against which to make the request.
+    :type project_id: str
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: str
+    """
+    template_fields = ("partial_keys",)
+
+    @apply_defaults
+    def __init__(
+        self,
+        partial_keys: List,
+        gcp_conn_id: str = 'google_cloud_default',
+        project_id: Optional[str] = None,
```

Comment:
It doesn't make much difference, but gcp_conn_id should be the last parameter. I would have ignored it, but now the order of the parameters doesn't match the documentation.